src/credix_lp/depository.ts

```diff
@@ -0,0 +1,286 @@
+
+const CREDIX_LP_INTERNAL_CREDIX_MARKETPLACE_NAMESPACE = 'credix-marketplace';
+const CREDIX_LP_INTERNAL_CREDIX_PASS_NAMESPACE = 'credix-pass';
+const CREDIX_LP_INTERNAL_PROGRAM_STATE_NAMESPACE = 'program-state';
+
+export class CredixLpDepository {
+  public constructor(
+    public readonly pda: PublicKey,
+    public readonly collateralMint: PublicKey,
+    public readonly collateralDecimals: number,
+    public readonly collateralSymbol: string,
+    public readonly depositoryCollateral: PublicKey,
+    public readonly depositoryShares: PublicKey,
+    public readonly credixProgramState: PublicKey,
+    public readonly credixGlobalMarketState: PublicKey,
+    public readonly credixSigningAuthority: PublicKey,
+    public readonly credixLiquidityCollateral: PublicKey,
+    public readonly credixSharesMint: PublicKey,
+    public readonly credixPass: PublicKey,
+    public readonly credixTreasuryCollateral: PublicKey,
+    public readonly credixMultisigKey: PublicKey,
+    public readonly credixMultisigCollateral: PublicKey,
+    public readonly credixProgramId: PublicKey
+  ) {}
+
+  public static async initialize({
+    connection,
+    uxdProgramId,
+    collateralMint,
+    collateralSymbol,
+    credixProgramId,
+  }: {
+    connection: Connection;
+    uxdProgramId: PublicKey;
+    collateralMint: PublicKey;
+    collateralSymbol: string;
+    credixProgramId: PublicKey;
+  }): Promise<CredixLpDepository> {
+    // First we need the credix global market state address
+    const credixGlobalMarketState = await this.findCredixGlobalMarketState(
+      credixProgramId
+    );
+    // Then the credix program state address
+    const credixProgramState = await this.findCredixProgramState(
+      credixProgramId
+    );
+
+    // Load credix IDL to be able to read onchain data
+    const provider = new AnchorProvider(
+      connection,
+      {} as Wallet,
+      AnchorProvider.defaultOptions()
+    );
+    const credixProgram = new Program<CredixIDL>(
+      credixIDL,
+      credixProgramId,
+      provider
+    );
+
+    // Then we read the content of the credixGlobalMarketState onchain
+    const credixGlobalMarketStateData =
+      await credixProgram.account.globalMarketState.fetchNullable(
+        credixGlobalMarketState
+      );
+    if (!credixGlobalMarketStateData) {
+      throw new Error('Could not read credixGlobalMarketState');
+    }
+    // Then we read the content of the credixProgramState onchain
+    const credixProgramStateData =
+      await credixProgram.account.programState.fetchNullable(
+        credixProgramState
+      );
+    if (!credixProgramStateData) {
+      throw new Error('Could not read credixProgramState');
+    }
+
+    // We now have all the data we need from the onchain credix configuration
+    const credixSharesMint = credixGlobalMarketStateData.lpTokenMint;
+    const credixTreasuryCollateral =
+      credixGlobalMarketStateData.treasuryPoolTokenAccount;
```

Comment:
nit, alternative syntax: 

```suggestion
    const {
       lpTokenMint: credixSharesMint,
       treasuryPoolTokenAccount: credixTreasuryCollateral,
    } = credixGlobalMarketStateData;
```